### DISCUSSION TITLE
Pydantic Settings: Load root BaseSettings model via environment variable as JSON-encoded string

### DISCUSSION THREAD

**Discussion by RafaelWO** [pydantic/pydantic#9692] [Question] [answered]
Hello,

I'm trying to load all settings via a single environment variable that contains the data as a JSON-encoded string. 

The [docs regarding parsing environment variables](https://docs.pydantic.dev/latest/concepts/pydantic_settings/#parsing-environment-variable-values) show an example where a nested sub-model is populated from such an environment variable.

However, I would like to load all settings from a single environment variable, e.g. like this:

```bash
# shell environment
# (not sure what the variable name must be to reference the root)
export SETTINGS='{"v0": "json-0", "sub_model": {"v1": "json-1", "v2": "json-2"}}'
```

```python
class SubModel(BaseModel):  
    v1: str
    v2: bytes


class Settings(BaseSettings):
    v0: str = "foo"
    sub_model: SubModel


print(Settings().model_dump())
```

Expected output:
```
{'v0': 'json-0, 'sub_model': {'v1': 'json-1', 'v2': b'json-2'}}
```

Is this currently possible?

**Comment by hramezani** [comment, marked as answer]
You need to define `v0` and `sub_model` env variables because they are the fields in the root model. You can define sub-model fields inside a json string.

So, if you want to have one root env, you need to define your model like:

```py
class SubModel(BaseModel):  
    v1: str
    v2: bytes


class Settings(BaseModel):
    v0: str = "foo"
    sub_model: SubModel


class RootSettings(BaseSettings):
    settings: Settings

print(RootSettings().model_dump())
```

**Reply by RafaelWO** [reply]
Thanks for providing the workaround! Any idea if the feature of loading the root settings (without a nested model) requires big changes?

Either way, I guess I can also expose the nested `settings` in the Python module to improve the developer experience and "hide" the nested model:

```python
class SubModel(BaseModel):  
    v1: str
    v2: bytes


class Settings(BaseModel):
    v0: str = "foo"
    sub_model: SubModel


class RootSettings(BaseSettings):
    settings: Settings


SETTINGS = RootSettings().settings
```
